### PR TITLE
Dev place page UI

### PR DESCRIPTION
**Pull request by dwnoble**
Updates dev place page to include:

* Topic tabs (Overview, Economics, etc)
* Place Overview panel that includes place overview text, map, and key demographics
* Child places link section
* Charts for selected categories

## Screenshots
Screenshot 2024-11-19 at 4 42 27 PM

Screenshot 2024-11-19 at 4 42 36 PM

## Still TODO:
* A better 404 page
* Only render the place overview panel if we have data to display in the panel
* The "Places in <place>" link section can get really large. Limit it to a maximum of "n" items
* Update chart section styles to match mockups
* Re-generate place summary text
* Merge /dev-place/ endpoint with /place/ endpoint
* Add experiment setup to only show the new place page for certain places but not others
* Add user engagement tracking
* Add i18n

Includes changes from https://github.com/datacommonsorg/website/pull/4738 . I'll rebase this PR once it's merged.

### CODE REVIEW

File: server/lib/fetch.py [hunk 1]
```diff
@@ -400,6 +400,45 @@ def property_values(nodes, prop, out=True, constraints=''):
...
 
 
+def multiple_property_values(nodes: List[str],
+                             props: List[str],
+                             out=True) -> Dict[str, Dict[str, List[str]]]:
+  """
+  Fetches specified properties for given nodes using the /v2/node API.
+
+  Args:
+      nodes (List[str]): List of node identifiers.
```
Comment (gmechali) [inline server/lib/fetch.py:410]: should wereplace "identifiers" by dcid just to be more specific?

File: server/routes/dev_place/utils.py [hunk 2]
```diff
@@ -232,60 +251,138 @@ def chart_config_to_overview_charts(chart_config):
...
     )
     if denominator:
       chart.denominator = denominator
+    if is_map_chart:
+      chart.childPlaceType = child_place_type
     charts.append(chart)
 
+    if page_config_item.get("relatedChart", {}).get("title"):
+      page_config_related_chart = page_config_item.get("relatedChart", {})
+      related_chart = Chart(
```
Comment (gmechali) [inline server/routes/dev_place/utils.py:273]: I might be missing it but what happens with this related_chart? Is this used somehwere?

**Review by gmechali — APPROVED** [review]

File: static/js/place/dev_place_main.tsx [hunk 3]
```diff
@@ -0,0 +1,499 @@
...
+ * Component that renders the header section of a place page.
+ * Displays the place name, category (if not Overview), and subheader text.
+ * Also shows the place DCID with a link to view it in the Knowledge Graph browser.
+ *
+ * @param props.category The current category being viewed
+ * @param props.place The place object containing name and DCID
+ * @param props.placeSubheader HTML string with additional place context
+ * @returns Header component for the place page
+ */
+const PlaceHeader = (props: {
```
Comment (beets) [inline static/js/place/dev_place_main.tsx:124]: would be nice if some of these components are moved into separate files

File: static/js/apps/explore/app.tsx [hunk 4]
```diff
@@ -220,7 +220,7 @@ export function App(props: { isDemo: boolean }): JSX.Element {
       isPendingRedirect = shouldSkipPlaceOverview(pageMetadata);
       if (isPendingRedirect) {
         const placeDcid = pageMetadata.place.dcid;
-        const url = `/place/${placeDcid}`;
+        const url = `/dev-place/${placeDcid}`;
```
Comment (beets) [inline static/js/apps/explore/app.tsx:223]: undo this until we are ready to launch. in the meantime, we could add another redirect in flask that does this based on environment

**Review by beets — APPROVED** [review]
ditto to gabe's comments, and a request to keep each component to a separate file if possible.

File: server/routes/dev_place/html.py [hunk 5]
```diff
@@ -28,19 +31,25 @@
...
 
   place_type_with_parent_places_links = utils.get_place_type_with_parent_places_links(
       place_dcid)
   place_names = place_api.get_i18n_name([place_dcid]) or {}
   place_name = place_names.get(place_dcid, place_dcid)
+  if g.locale == DEFAULT_LOCALE:
+    place_summary = get_place_summaries(place_dcid).get(place_dcid,
+                                                        {}).get("summary", "")
+  else:
+    place_summary = ""
```
[the remark this answers is not in the document]
Comment (dwnoble) [reply server/routes/dev_place/html.py]: Yeah that's right. Added a comment

File: static/js/apps/explore/app.tsx (hunk 4 above)
Comment (dwnoble) [reply static/js/apps/explore/app.tsx:223]: Good catch- reverted

File: server/tests/routes/api/dev_place_test.py [hunk 6]
```diff
@@ -119,16 +122,41 @@ def test_dev_place_charts(self, mock_obs_point, mock_property_values,
...
-        if prop == 'typeOf':
-          return {dcid: ['Country'] for dcid in dcids}
-        elif prop == 'name':
-          return {dcid: ['United States'] for dcid in dcids}
-        else:
-          return {dcid: [] for dcid in dcids}
+      # Mock obs_point_within for finding child places existence check for map-based stat vars
+      mock_obs_point_within.return_value = {
+          "byVariable": {
+              "Count_CriminalActivities_CombinedCrime": {
```
[the remark this answers is not in the document]
Comment (dwnoble) [reply server/tests/routes/api/dev_place_test.py]: Good call. Moved these to mock_data

File: server/routes/dev_place/utils.py (hunk 2 above)
Comment (dwnoble) [reply server/routes/dev_place/utils.py]: Good catch. Appended this chart to the `charts` list.

File: server/lib/fetch.py (hunk 1 above)
Comment (dwnoble) [reply server/lib/fetch.py]: fixed

File: static/js/place/dev_place_main.tsx (hunk 3 above)
Comment (dwnoble) [reply static/js/place/dev_place_main.tsx:159]: I'll address this in a follow-up. I like to have some of the smaller components in the same file, but agree this file is also getting pretty large.

**Pull request merged**